Donate Help Page FA Form basic rendering

danielfmiranda opened the pull request:
# Description

<!-- Describe the PR here -->

Link to sample test page: https://donate-foundation-s-11173-dona-jtqs8w.mofostaging.net/en/help/
Related PRs/issues: #11173


This PR handles the basic rendering of the unstyled FA form.


# Screenshots

![Screenshot 2023-10-25 at 20-13-25 Donate Help](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/fa76733c-5e20-4c45-8e1a-8ed7ae3125f7)

mmmavis reviewed (changes requested):
Hey @danielfmiranda , 

At a glance it's not possible to tell what code is copied from FormAssembly vs our own custom code so it's tricky to do code review. Could you mimic what we have for petition form ([formassembly_body.html](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/formassembly_body.html#L3-L48) and [formassembly_head.html](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/formassembly_head.html#L3-L26)) and add notes to the top of the two FA related files? Having notes/instructions will also help future code edits from breaking the form.

We'll also need to edit default CSP values on various files for recaptcha to work!

Thanks!

danielfmiranda commented on network-api/networkapi/templates/donate/fragments/formassembly_body.html:
```diff
@@ -0,0 +1,168 @@
+                    return;
+                }
+
+    // Retrieve the url, and attempt to get the first parameter (locale)
+                const url = new URL(window.location);
+                const locale = url.pathname.split('/')[1] || null;
+
+    // If locale does not exist in the url, then there's nothing more to do.
+                if (!locale) {
+                    return;
+                }
+
+    // Match up locale codes with the locale dropdown entries
+                const localeMap = {
+                    'zh-CN': 'tfa_218',
+                    'zh-TW': 'tfa_218',
+                    'cs': 'tfa_219',
+                    'da': 'tfa_220',
+                    'nl': 'tfa_221',
+                    'en-CA': 'tfa_222',
+                    'en-GB': 'tfa_222',
+                    'en-US': 'tfa_222',
+                    'en-ZA': 'tfa_222',
+                    'fr': 'tfa_223',
+                    'de': 'tfa_224',
+                    'it': 'tfa_225',
+                    'ja': 'tfa_226',
+                    'pl': 'tfa_228',
+                    'pt-BR': 'tfa_229',
+                    'ru': 'tfa_230',
+                    'es-MX': 'tfa_231',
+                    'es-AR': 'tfa_231',
+                    'es-ES': 'tfa_231',
+                    'es-CL': 'tfa_231'
+                }
+
+    // Assign the locale, or default to 'Other' if the locale isn't supported.
+                varLocale.value = localeMap[locale] || 'tfa_227';
+            });
+
```
Hi @mmmavis just calling out this custom localization JS that I found from Mia and the thunderbird team. I am probably going to have to update this in my localization PR as they are referencing locales we do not use on the foundation site (ex: en-US, es-MX, etc.).

danielfmiranda commented:
Hi @mmmavis! Thanks for the review. 

I have updated this PR with what we discussed on slack which includes:
- Updated env.default to include the URLs needed to render the form / recaptcha
- Some comments at the top of the files, which will be updated as I make more changes to the form such as styling/localization


This pr is now ready for re-review whenever you find a free moment. Thanks!

mmmavis reviewed (changes requested):
Looking great! If we can update CSP configs in `app.json` and `continous-integration.yml` as well, that'd be rad 💯

danielfmiranda commented:
Thanks for the review @mmmavis! Just updated both `app.json` and `continuous-integration.yml` 👍

mmmavis reviewed (approved):
💯
